anyway/models.py

```diff
@@ -871,13 +872,54 @@ class NewsFlash(Base):
     street2_hebrew = Column(Text(), nullable=True)
     non_urban_intersection_hebrew = Column(Text(), nullable=True)
     road_segment_name = Column(Text(), nullable=True)
+    critical = Column(Boolean(), nullable=True)
     newsflash_location_qualification = Column(
         Integer(),
         nullable=False,
         server_default=text(f"{NewsflashLocationQualification.NOT_VERIFIED.value}"),
     )
     location_qualifying_user = Column(BigInteger(), nullable=True)
 
+    def __init__(self, **kwargs):
+        super(NewsFlash, self).__init__(**kwargs)
+        self.critical = self.get_critical()
+
+    def get_critical(
+        self,
+        years_before=5,
+        suburban_road_severe_value=10,
+        suburban_road_killed_value=3,
+        urban_severe_value=2,
+    ):
+        from anyway.widgets.suburban_widgets.injured_count_by_severity_widget import (
+            InjuredCountBySeverityWidget,
+        )
+        from anyway.request_params import get_latest_accident_date
+
+        if self.road1 is None or self.road_segment_name is None:
+            return None
+        last_accident_date = get_latest_accident_date(table_obj=AccidentMarkerView, filters=None)
+        end_time = last_accident_date.to_pydatetime().date()
+        start_time = datetime.date(end_time.year + 1 - years_before, 1, 1)
+        critical_values = InjuredCountBySeverityWidget.get_injured_count_by_severity(
+            self.road1, self.road_segment_name, start_time, end_time
+        )
+        if critical_values == {}:
+            return None
+        critical = None
+        resolution = BE_CONST.ResolutionCategories(self.resolution)
+        if resolution == BE_CONST.ResolutionCategories.SUBURBAN_ROAD:
+            critical = (
+                (critical_values["severe_injured_count"] / suburban_road_severe_value)
+                + (critical_values["killed_count"] / suburban_road_killed_value)
+            ) >= 1
+        elif resolution == BE_CONST.ResolutionCategories.STREET:
+            critical = (
+                (critical_values["severe_injured_count"] / urban_severe_value)
+                + critical_values["killed_count"]
+            ) >= 1
+        return critical
```

Comment:
When `road1` and `road_segment` are not None, resolution will not be STREET. Anyway, even if it is, the `critical_values` here are calculated from `road1` and `road_segment`, and thus are not relevant to STREET locations.

Reply:
you right, i will talk with atalya on this issue and change the code after